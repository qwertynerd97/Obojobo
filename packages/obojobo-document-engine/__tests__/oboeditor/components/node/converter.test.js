jest.mock('Common', () => ({
	Store: {
		getItemForType: () => ({
			slateToObo: jest.fn(),
			oboToSlate: jest.fn()
		})
	}
}))

import Converter from 'src/scripts/oboeditor/components/node/converter'
const BREAK_NODE = 'ObojoboDraft.Chunks.Break'

describe('Component converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			nodes: [
				{
					type: BREAK_NODE,
					data: { get: () => ({ width: 'large' }) }
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType'
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
