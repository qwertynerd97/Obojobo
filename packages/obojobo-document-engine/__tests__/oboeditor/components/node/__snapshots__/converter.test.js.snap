// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "nodes": Array [
    undefined,
  ],
  "object": "block",
  "type": "oboeditor.component",
}
`;

exports[`Component converter slateToObo converts a Slate node to an OboNode with content 1`] = `undefined`;
